src/main/kotlin/hwr/oop/projects/peakpoker/core/hand/HandEvaluator.kt

```diff
@@ -1,16 +1,88 @@
 package hwr.oop.projects.peakpoker.core.hand
 
-import hwr.oop.projects.peakpoker.core.card.CommunityCards
+import hwr.oop.projects.peakpoker.core.card.Card
 import hwr.oop.projects.peakpoker.core.card.HoleCards
+import hwr.oop.projects.peakpoker.core.card.CommunityCards
+import hwr.oop.projects.peakpoker.core.card.Rank
+import hwr.oop.projects.peakpoker.core.hand.HandRank
 
 object HandEvaluator {
-    internal fun evaluate(hand: HoleCards, communityCards: CommunityCards): HandRank {
-        TODO("Combine hole cards and community cards, evaluate the hand, and return the HandRank")
+
+    /**
+     * Evaluates the best HandRank from exactly five cards.
+     */
+    internal fun evaluate(cards: List<Card>): HandRank {
+        require(cards.size == 5) { "Hand must contain exactly 5 cards" }
+        require(cards.distinct().size == 5) { "Hand must contain 5 unique cards" }
+
+        val ranks = cards.map { it.rank.ordinal }.sorted()
+        val suits = cards.map { it.suit }
+        val rankCounts = ranks.groupingBy { it }.eachCount().values.sortedDescending()
+        val isFlush = suits.distinct().size == 1
+        val isStraight = when {
+            ranks.zipWithNext().all { (a, b) -> b == a + 1 } -> true
+            /*
+            * After sorting the card list we create a series of pairs (1,2,3,4,5) -> ((1,2),(2,3),...,(4,5)) and evaluating
+            * if their pair member inside the brackets is the successor -exactly the next card.If this property holds for all
+            * pairs in the zip, we have a straight, with the edge case of ordinal 14, the Ace, listed in Line 25.*/
+            //Special Case: Ace can be low in a straight
```

Comment:
Let us make this comment more concise to ensure clean code (and less lines ;) )
```suggestion
            // After sorting, pairs of consecutive ranks (e.g., (1,2), (2,3), ...) are checked to confirm each card is a direct successor. This determines a straight, with a special case for Ace (ordinal 14) as the lowest card.
```